pkg/build/builder/docker.go

```diff
@@ -117,18 +119,17 @@ func (d *DockerBuilder) Build() error {
 
 			timing.RecordNewStep(ctx, buildapiv1.StagePullImages, buildapiv1.StepPullBaseImage, startTime, metav1.Now())
 
-			if err != nil {
+			if err != nil && !stageNames.Has(imageName) {
```

Comment:
i don't understand this. please add a comment explaining what it's doing and why you are ignoring the error in some cases.

Reply:
Comment added